doc/install.rst

```diff
@@ -1,58 +1,655 @@
+
+MPI
+---
+
+Arbor uses MPI for distributed systems. By default it is built without MPI support, which
+can enabled by setting the ``DARB_DISTRIBUTED_MODEL`` CMake parameter.
+An example of building Arbor with MPI, high-performance threading and optimizations enabled
+is:
+
+.. code-block:: bash
+
+    # set the compiler wrappers
+    export CC=`which mpicc`
+    export CXX=`which mpicxx`
+
+    # configure with mpi, tbb threading and compiled with optimizations
+    cmake .. -DARB_DISTRIBUTED_MODEL=mpi \      # Use MPI
+             -DCMAKE_BUILD_TYPE=release  \      # Optimizations on
+             -DARB_THREADING_MODEL=tbb   \      # TBB threading library
+
+    # run unit tests for global communication on 2 MPI ranks
+    mpirun -n 2 ./tests/global_communication.exe
+
+The first step to building with MPI support is to set the ``CC`` and ``CXX`` environment variables to refer to the mpi compiler wrappers.
+
+.. Note::
+    MPI distributions provide **compiler wrappers** for compiling MPI applications.
+
+    In the example above the compiler wrappers for C and C++ called
+    ``mpicc`` and ``mpicxx`` respectively. The name of the compiler wrapper
+    is dependent on the MPI distribution.
+
+    The wrapper forwards the compilation to a compiler, like GCC, and
+    you have to ensure that this compiler is able to compile Arbor. For wrappers
+    that call GCC, Intel or Clang compilers, you can pass the ``--version`` flag
+    to the wrapper. For example, on a Cray system where the C++ wrapper is called ``CC``:
+
+    .. code-block:: bash
+
+        $ CC --version
+        g++ (GCC) 6.2.0 20160822 (Cray Inc.)
+
+Cray Systems
+------------
+
+The compiler used by the MPI wrappers is set using a "programming enviroment" module.
+The first thing to do is change this module, which by default is set to the Cray
+programming environment.
+For example, to use the GCC compilers, select the GNU programming enviroment:
+
+.. code-block:: bash
+
+    module swap PrgEnv-cray PrgEnv-gnu
+
+The version of the GCC can then be set by choosing an appropriate gcc module.
+In the example below we use ``module avail`` to see which versions of GCC are available,
+then choose GCC 7.1.0
+
+.. code-block:: bash
+
+    $ module avail gcc      # see all available gcc versions
+
+    ------------------------- /opt/modulefiles ---------------------------
+    gcc/4.9.3    gcc/6.1.0    gcc/7.1.0    gcc/5.3.0(default)    gcc/6.2.0
+
+    $ module swap gcc/7.1.0 # swap gcc 5.3.0 for 7.1.0
+
+    $ CC --version          # test that the wrapper uses gcc 7.1.0
+    g++ (GCC) 7.1.0 20170502 (Cray Inc.)
+
+    # set compiler wrappers
+    $ export CC=`which cc`
+    $ export CXX=`which CC`
+
+Note that the C and C++ compiler wrappers are called ``cc`` and ``CC``
+respectively on Cray systems.
+
+CMake detects that it is being run in the Cray programming environment, which makes
+our lives a little bit more difficult (CMake sometimes tries a bit too hard to help).
```

Comment:
Dry... :smile:

Reply:
Good thing I wrote this instead of @halfflat.